### PR TITLE
Modified contexts for the PlayerLogin event.

### PR DESCRIPTION
**Pull request by heypr**
- The server_hostname context now returns the server address that the player used to connect to the server.

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/events/player/PlayerLoginScriptEvent.java [hunk 1]
```diff
@@ -28,7 +28,8 @@ public class PlayerLoginScriptEvent extends BukkitScriptEvent implements Listene
     // @Warning Generally avoid this event. This is not a way to get a 'first join' event. This is an internal technical event, with specific uses (eg custom whitelisting).
     //
     // @Context
-    // <context.hostname> returns an ElementTag of the player's hostname.
+    // <context.hostname> returns an ElementTag of the server address that the player used to connect to the server.
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/events/player/PlayerLoginScriptEvent.java]: You can't just casually swap a pre-existing context like this, that will break any existing scripts.

Comment (heypr) [reply plugin/src/main/java/com/denizenscript/denizen/events/player/PlayerLoginScriptEvent.java]: Fixed.

File: plugin/src/main/java/com/denizenscript/denizen/events/player/PlayerLoginScriptEvent.java [hunk 2]
```diff
@@ -28,7 +28,9 @@ public class PlayerLoginScriptEvent extends BukkitScriptEvent implements Listene
     // @Warning Generally avoid this event. This is not a way to get a 'first join' event. This is an internal technical event, with specific uses (eg custom whitelisting).
     //
     // @Context
-    // <context.hostname> returns an ElementTag of the player's hostname.
+    // <context.hostname> returns an ElementTag of the player's IP address.
+    // <context.server_hostname> returns an ElementTag of the server address that the player used to connect to the
+    // server.
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/events/player/PlayerLoginScriptEvent.java]: this random newline split shouldn't be here

Comment (heypr) [reply plugin/src/main/java/com/denizenscript/denizen/events/player/PlayerLoginScriptEvent.java]: Fixed.

File: plugin/src/main/java/com/denizenscript/denizen/events/player/PlayerLoginScriptEvent.java [hunk 3]
```diff
@@ -77,10 +78,12 @@ public ScriptEntryData getScriptEntryData() {
...
-        if (name.equals("hostname")) {
-            return new ElementTag(event.getAddress().toString());
-        }
-        return super.getContext(name);
+        return switch (name) {
+            case "hostname" -> new ElementTag(event.getAddress().toString());
+            case "server_hostname" -> new ElementTag(event.getHostname());
+            default -> super.getContext(name);
+        };
+
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/events/player/PlayerLoginScriptEvent.java]: stray newline

Comment (heypr) [reply plugin/src/main/java/com/denizenscript/denizen/events/player/PlayerLoginScriptEvent.java]: Removed.

**Pull request merged**